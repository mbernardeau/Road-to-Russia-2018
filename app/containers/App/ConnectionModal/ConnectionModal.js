import React, {
    Component,
} from 'react';

import firebase from 'firebase';
import RaisedButton from 'material-ui/RaisedButton';
import FaGoogle from 'react-icons/lib/fa/google';

import styles from './ConnectionModal.scss';

class ConnectionModal extends Component {
  firebaseAuthPopup() {
    const provider = new firebase.auth.GoogleAuthProvider();
    firebase.auth().signInWithPopup(provider);
  }

  render() {
    return (
      <div className={styles.connectionContainer}>
        <RaisedButton
          label="Connexion avec Google"
          labelPosition="after"
          primary
          onClick={this.firebaseAuthPopup}
          icon={<FaGoogle />}
        />
      </div>
    );
  }
}

export default ConnectionModal;
