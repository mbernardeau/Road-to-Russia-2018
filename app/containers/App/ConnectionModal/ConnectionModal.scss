.connectionContainer {
    display: flex;
    flex-direction: column;
    align-items: center;
}