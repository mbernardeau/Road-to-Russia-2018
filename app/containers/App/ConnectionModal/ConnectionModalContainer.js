import { withFirebase } from 'react-redux-firebase';

import ConnectionModal from './ConnectionModal';

export default withFirebase(ConnectionModal);
