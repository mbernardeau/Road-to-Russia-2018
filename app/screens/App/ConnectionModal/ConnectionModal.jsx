import React, { Component, Fragment } from 'react'
import PropTypes from 'prop-types'

import Button from 'material-ui/Button'
import FaGoogle from 'react-icons/lib/fa/google'
import FaFacebook from 'react-icons/lib/fa/facebook'
import { DialogContent, DialogTitle } from 'material-ui/Dialog'

import './ConnectionModal.scss'

class ConnectionModal extends Component {
  authenticateWithGoogle = () => {
    this.props.firebase.login({
      provider: 'google',
      type: 'popup',
    })
  }

  authenticateWithFacebook = () => {
    this.props.firebase.login({
      provider: 'facebook',
      type: 'popup',
    })
  }

  render() {
    return (
      <Fragment>
        <DialogTitle>Connexion</DialogTitle>
        <DialogContent className="auth-btns-container">
          <Button color="primary" onClick={this.authenticateWithGoogle} variant="raised">
            <FaGoogle />&nbsp; Connexion avec Google
          </Button>

          <Button color="secondary" onClick={this.authenticateWithFacebook} variant="raised">
            <FaFacebook />&nbsp; Connexion avec Facebook
          </Button>
        </DialogContent>
      </Fragment>
    )
  }
}

ConnectionModal.propTypes = {
  firebase: PropTypes.shape({
    login: PropTypes.func.isRequired,
  }).isRequired,
}

export default ConnectionModal
