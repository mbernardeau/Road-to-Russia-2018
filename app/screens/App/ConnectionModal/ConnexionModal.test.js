import React from 'react'
import renderer from 'react-test-renderer'

import ConnectionModal from './ConnectionModal'

describe('screens/App/ConnectionModal', () => {
  const props = {
    firebase: {
      login: () => {},
    },
  }

  it('renders correctly', () => {
    const tree = renderer.create(<ConnectionModal {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
