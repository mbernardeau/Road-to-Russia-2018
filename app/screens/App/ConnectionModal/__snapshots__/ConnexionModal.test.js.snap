// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screens/App/ConnectionModal renders correctly 1`] = `
Array [
  <DialogTitle>
    Connexion
  </DialogTitle>,
  <DialogContent
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "padding": 15,
      }
    }
  >
    <Button
      color="primary"
      onClick={[Function]}
      style={
        Object {
          "marginBottom": 10,
        }
      }
      variant="raised"
    >
      <FaGoogle />
        Connexion avec Google
    </Button>
    <Button
      color="accent"
      onClick={[Function]}
      style={
        Object {
          "marginBottom": 10,
        }
      }
      variant="raised"
    >
      <FaFacebook />
        Connexion avec Facebook
    </Button>
  </DialogContent>,
]
`;
